src/config.py

```diff
@@ -1,26 +1,80 @@
-_image_generator_configs = [
-    {
-        "build_args": {
-            "TAG_FOR_BASE_MICROMAMBA_IMAGE": "jammy-cuda-11.8.0",
-            "CUDA_MAJOR_MINOR_VERSION": "11.8",  # Should match the previous one.
-            "ENV_IN_FILENAME": "gpu.env.in",
-            "ARG_BASED_ENV_IN_FILENAME": "gpu.arg_based_env.in",
+_image_generator_configs = {
+    0: [
+        {
+            "build_args": {
+                "TAG_FOR_BASE_MICROMAMBA_IMAGE": "jammy-cuda-11.8.0",
+                "CUDA_MAJOR_MINOR_VERSION": "11.8",  # Should match the previous one.
+                "ENV_IN_FILENAME": "gpu.env.in",
+                "ARG_BASED_ENV_IN_FILENAME": "gpu.arg_based_env.in",
+            },
+            "additional_packages_env_in_file": "gpu.additional_packages_env.in",
+            "image_tag_generator": "{image_version}-gpu",
+            "env_out_filename": "gpu.env.out",
+            "pytest_flags": ["--use-gpu"],
+            "image_type": "gpu",
         },
-        "additional_packages_env_in_file": "gpu.additional_packages_env.in",
-        "image_tag_generator": "{image_version}-gpu",
-        "env_out_filename": "gpu.env.out",
-        "pytest_flags": ["--use-gpu"],
-        "image_type": "gpu",
-    },
-    {
-        "build_args": {
-            "TAG_FOR_BASE_MICROMAMBA_IMAGE": "jammy",
-            "ENV_IN_FILENAME": "cpu.env.in",
+        {
+            "build_args": {
+                "TAG_FOR_BASE_MICROMAMBA_IMAGE": "jammy",
+                "ENV_IN_FILENAME": "cpu.env.in",
+            },
+            "additional_packages_env_in_file": "cpu.additional_packages_env.in",
+            "image_tag_generator": "{image_version}-cpu",
+            "env_out_filename": "cpu.env.out",
+            "pytest_flags": [],
+            "image_type": "cpu",
         },
-        "additional_packages_env_in_file": "cpu.additional_packages_env.in",
-        "image_tag_generator": "{image_version}-cpu",
-        "env_out_filename": "cpu.env.out",
-        "pytest_flags": [],
-        "image_type": "cpu",
-    },
-]
+    ],
+    1: [
+        {
+            "build_args": {
+                "TAG_FOR_BASE_MICROMAMBA_IMAGE": "jammy-cuda-12.4.1",
+                "CUDA_MAJOR_MINOR_VERSION": "12",  # Should match the previous one.
```

Comment:
Did we update CUDA version for 1.x major versions?

Reply:
Should we specify the minor version (rather than just 12)

Reply:
Sorry this was leftover from testing; have fixed. the v2 cuda config is below